spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Optimizer.scala

```diff
@@ -284,6 +286,33 @@ abstract class Optimizer[T: ClassTag, D](
   }
 
   def prepareInput(): Unit = {}
+
+  /**
+   * Set constant gradient clipping
+   * @param min the minimum value to clip by
+   * @param max the maximum value to clip by
+   * @return
+   */
+  def setConstantGradientClipping(min: Float, max: Float)
+  : this.type = {
+    require(min < max, "min value must be smaller than max")
+    gradientClippingParams.enableConstantClipping = true
+    gradientClippingParams.minValueClip = min
+    gradientClippingParams.maxValueClip = max
+    this
+  }
+
+  /**
+   * Clip gradient to a maximum L2-norm
+   * @param clipNorm gradient L2-Norm threshold
+   * @return
+   */
+  def setGradientClippingByNorm(clipNorm: Float)
```

Comment:
method name should include L2, because there're several kinds of norms (e.g. l1)

Reply:
Updated